python/ray/experimental/actor_pool.py

```diff
@@ -0,0 +1,238 @@
+    def __init__(self, actors):
+        """Create an Actor pool from a list of existing actors.
+
+        An actor pool is a utility class similar to multiprocessing.Pool that
+        lets you schedule Ray tasks over a fixed pool of actors.
+
+        Arguments:
+            actors (list): List of Ray actor handles to use in this pool.
+
+        Examples:
+            >>> a1, a2 = Actor.remote(), Actor.remote()
+            >>> pool = ActorPool([a1, a2])
+            >>> print(pool.map(lambda a, v: a.double.remote(v), [1, 2, 3, 4]))
+            [2, 4, 6, 8]
+        """
+        self._idle_actors = list(actors)
+        self._future_to_actor = {}
+        self._index_to_future = {}
+        self._next_task_index = 0
+        self._next_return_index = 0
+        self._pending_submits = []
+
+    def map(self, fn, values):
+        """Apply the given function in parallel over the actors and values.
+
+        This returns an ordered iterator that will return results of the map
+        as they finish. Note that you must iterate over the iterator to force
+        the computation to finish.
+
+        Arguments:
+            fn (func): Function that takes (actor, value) as argument and
+                returns an ObjectID computing the result over the value. The
+                actor will be considered busy until the ObjectID completes.
+            values (list): List of values that fn(actor, value) should be
+                applied to.
+
+        Returns:
+            Iterator over results from applying fn to the actors and values.
+
+        Examples:
+            >>> pool = ActorPool(...)
+            >>> print(pool.map(lambda a, v: a.double.remote(v), [1, 2, 3, 4]))
+            [2, 4, 6, 8]
+        """
+        for v in values:
+            self.submit(fn, v)
+        while self.has_next():
+            yield self.get_next()
+
+    def map_unordered(self, fn, values):
+        """Similar to map(), but returning an unordered iterator.
+
+        This returns an unordered iterator that will return results of the map
+        as they finish. This can be more efficient that map() if some results
+        take longer to compute than others.
+
+        Arguments:
+            fn (func): Function that takes (actor, value) as argument and
+                returns an ObjectID computing the result over the value. The
+                actor will be considered busy until the ObjectID completes.
+            values (list): List of values that fn(actor, value) should be
+                applied to.
+
+        Returns:
+            Iterator over results from applying fn to the actors and values.
+
+        Examples:
+            >>> pool = ActorPool(...)
+            >>> print(pool.map(lambda a, v: a.double.remote(v), [1, 2, 3, 4]))
+            [6, 2, 4, 8]
+        """
+        for v in values:
+            self.submit(fn, v)
+        while self.has_next():
+            yield self.get_next_unordered()
+
+    def submit(self, fn, value):
```

Comment:
`submit` or `apply`?

Reply:
multiprocessing does apply https://docs.python.org/2/library/multiprocessing.html#multiprocessing.pool.multiprocessing.Pool.apply

Reply:
So the semantics are a little different from apply since we don't return a future. This is actually a pretty fundamental limitation of Ray in that you need to schedule the task to have a future (similar to the Ray serve issues).